core/src/main/java/io/grpc/internal/ReadableBuffers.java

```diff
@@ -312,7 +312,8 @@ public int arrayOffset() {
   /**
    * An {@link InputStream} that is backed by a {@link ReadableBuffer}.
    */
-  private static class BufferInputStream extends InputStream implements KnownLength {
+  private static final class BufferInputStream extends InputStream
+      implements KnownLength, ReadableBuffer {
```

Comment:
Do we actually need all of the entire `ReadableBuffer` interface here? Seems like we really only use the copy method.